runtime/oti/j9nonbuilder.h

```diff
@@ -4070,11 +4070,20 @@ typedef struct J9InternalHookRecord {
 	struct J9Pool *instanceObjects;
 } J9InternalHookRecord;
 
+typedef struct J9DelayedSingleThreadModeOpRecord {
+	jobject globalObjectRef;
+	UDATA operation;
```

Comment:
Does pool iteration guarantee the same order FIFO iteration order?  i.e. the first notify we delay should be the first we do after exiting single threaded mode.  Similarly the 2nd should be the 2nd, etc.

If the pool iteration order isn't guaranteed (and documented!) I think these should have a `J9DelayedSingleThreadModeOpRecord *next` field as well and they should be stored in a linked list.  The pool is fine for backing the allocation but we should iterate through the list to do the notifies.

And likely need two pointers - one for the head of the list and one for the tail so we can insert quickly at the tail and iterate from the head